plugins/destination/elasticsearch/client/write.go

```diff
@@ -78,6 +78,7 @@ func (c *Client) writeRecord(ctx context.Context, table *schema.Table, record ar
 	resp, err := c.client.Bulk(bytes.NewReader(buf.Bytes()),
 		c.client.Bulk.WithContext(ctx),
 		c.client.Bulk.WithIndex(index),
+		c.client.Bulk.WithRefresh("wait_for"),
```

Comment:
required to flush the data before the write returns (either this or we do the refresh before the delete stale)